django/contrib/auth/tokens.py

```diff
@@ -41,19 +41,33 @@ def check_token(self, user, token):
         if not constant_time_compare(self._make_token_with_timestamp(user, ts), token):
             return False
 
-        # Check the timestamp is within limit. Timestamps are rounded to
+        # Check the timestamp is within limit.
+        # If using PASSWORD_RESET_TIMEOUT_DAYS, Timestamps are rounded to
         # midnight (server time) providing a resolution of only 1 day. If a
         # link is generated 5 minutes before midnight and used 6 minutes later,
-        # that counts as 1 day. Therefore, PASSWORD_RESET_TIMEOUT_DAYS = 1 means
-        # "at least 1 day, could be up to 2."
-        if (self._num_days(self._today()) - ts) > settings.PASSWORD_RESET_TIMEOUT_DAYS:
+        # that counts as 1 day. Therefore, PASSWORD_RESET_TIMEOUT_DAYS = 1
+        # means "at least 1 day, could be up to 2".
+        # Because settings may be changed between
+        # PASSWORD_RESET_TIMEOUT and PASSWORD_RESET_TIMEOUT_DAYS,
+        # convert both to seconds to compare.
+        if hasattr(settings, 'PASSWORD_RESET_TIMEOUT'):
```

Comment:
This will always be `True` given `PASSWORD_RESET_TIMEOUT` is defined in `global_settings`.

Reply:
@charettes Thanks for the comment.
I removed `PASSWORD_RESET_TIMEOUT` from `global_settings`. I think we can add it again when the `PASSWORD_RESET_TIMEOUT_DAYS` removed.

Reply:
I think we should leave `PASSWORD_RESET_TIMEOUT` in a `global_settings.py` and always use `PASSWORD_RESET_TIMEOUT` with a warning when both are modified. Currently you can set both what is confusing (it's not clear which one will be used), this would also simplify `PasswordResetTokenGenerator` and process of removing  `PASSWORD_RESET_TIMEOUT_DAYS` in Django 4.0. We could do sth like:
```diff
diff --git a/django/conf/__init__.py b/django/conf/__init__.py                                                                                         
index 510260098e..8088f65701 100644                                                                                                                    
--- a/django/conf/__init__.py                                                                                                                          
+++ b/django/conf/__init__.py                                                                                                                          
@@ -145,6 +145,12 @@ class Settings:                                                                                                                   
             raise ImproperlyConfigured("The SECRET_KEY setting must not be empty.")                                                                   
                                                                                                                                                       
         if self.is_overridden('PASSWORD_RESET_TIMEOUT_DAYS'):                                                                                         
+            if self.is_overridden('PASSWORD_RESET_TIMEOUT'):                                                                                          
+                raise ImproperlyConfigured(                                                                                                           
+                    'PASSWORD_RESET_TIMEOUT_DAYS/PASSWORD_RESET_TIMEOUT are '                                                                         
+                    'mutually exclusive.'                                                                                                             
+                )                                                                                                                                     
+            setattr(self, 'PASSWORD_RESET_TIMEOUT', self.PASSWORD_RESET_TIMEOUT_DAYS * 60 * 60 * 24)                                                  
             warnings.warn(PASSWORD_RESET_TIMEOUT_DAYS_DEPRECATED_MSG, RemovedInDjango40Warning)                                                       
                                                                                                                                                       
         if hasattr(time, 'tzset') and self.TIME_ZONE:                                                                                                 
@@ -191,6 +197,7 @@ class UserSettingsHolder:                                                                                                          
     def __setattr__(self, name, value):                                                                                                               
         self._deleted.discard(name)                                                                                                                   
         if name == 'PASSWORD_RESET_TIMEOUT_DAYS':                                                                                                     
+            setattr(self, 'PASSWORD_RESET_TIMEOUT', value * 60 * 60 * 24)                                                                             
             warnings.warn(PASSWORD_RESET_TIMEOUT_DAYS_DEPRECATED_MSG, RemovedInDjango40Warning)                                                       
         super().__setattr__(name, value)                                                                                                              
                                                                                                                                                       
```

Reply:
@hramezani :point_up:

Reply:
@felixxm, `PASSWORD_RESET_TIMEOUT` added to `global_settings.py`.
Also, I added a test for the `ImproperlyConfigured`. and change `PasswordResetTokenGenerator` to just handle the `PASSWORD_RESET_TIMEOUT`.